### DISCUSSION TITLE
Basic Auth or Authorization Header Bearer Token

### DISCUSSION THREAD

**Discussion by michaelwechner** [amithkoujalgi/ollama4j#15] [Q&A] [answered]
Do you have plans to support

- Either Basic Auth
- Or an Authorization Header Bearer Token

?

I understand Ollama does not support this, but when using a ReverseProxy in front of Ollama, then this would help a lot re security :-)

Thanks

Michael

**Comment by michaelwechner** [comment, marked as answer]
I have added a PR https://github.com/amithkoujalgi/ollama4j/pull/12

**Reply by amithkoujalgi** [reply]
Thanks @michaelwechner! I've merged the PR.

**Comment by amithkoujalgi** [comment]
This would certainly help in scenarios where Ollama would be setup behind a gateway/reverse proxy with basic auth.

I believe there could be more auth types needed as well. Also, I think its also a good idea to provide capability to set custom headers to the requests.
What's your take on this?
